tempodb/tempodb.go

```diff
@@ -351,7 +351,7 @@ func (rw *readerWriter) Find(ctx context.Context, tenantID string, id common.ID,
 		return foundObject, meta.DataEncoding, nil
 	})
 
-	return partialTraces, dataEncodings, err
+	return partialTraces, dataEncodings, stats.ErrsCount(), err
```

Comment:
the way this is written these block errors get swallowed and are no longer logged. i think we should
- ditch jobStats
- have pool.RunJobs() return partialTraces, dataEncodings, multierror, error
  - multierror are function errors, error is a fatal error
- tempodb.Find also returns multierror and error
- querier.FindTraceByID
  - fails if the fatal error is returned like normal
  - multierror.len > 0 then log them
  - pass multierror.count back to the query frontend

Reply:
That's a good point. I've changed the methods to pass the errors up to the querier, so they're log in there too.